plugins/playstation/src/playstation-header.vala

```diff
@@ -0,0 +1,87 @@
+// This file is part of GNOME Games. License: GPLv3
+
+private class Games.PlayStationHeader : Object {
+  private File cue_file;
+  private File bin_file;
+	private string parent_directory;
+
+	private const size_t[] OFFSETS = { 0xD368, 0xD3A8 };
+	private const string[] IDS = { "SLUS", "SCUS", "SLES", "SCES", "SLPS", "SLPM", "SCPS" };
+
+	private size_t? header_offset;
+
+  public PlayStationHeader (File cue_file) {
```

Comment:
You should build it from the binary file, this has nothing to do with cue sheets.